ml_task_runner.py

```diff
@@ -0,0 +1,135 @@
+import json
+import os
+import signal
+import sys
+import time
+
+import backoff
+import nbformat
+import nbconvert
+
+from api_clients.task import TaskClient
+from api_clients.core import CoreClient
+
+class MLTaskRunner(object):
+    shutting_down = False
+    download_complete = False
+
+    def __init__(self, configuration):
+        self.configuration = configuration
+        self.task_client = TaskClient(configuration['services']['task-service']['base_url'],
+                                      configuration['auth_token'],
+                                      configuration['services']['task-service']['worker_id'])
+        self.core_client = CoreClient(configuration['services']['core-service']['base_url'],
+                                      configuration['auth_token'],
+                                      configuration['services']['task-service']['worker_id'])
+        self.task = None
+
+    @staticmethod
+    def run_notebook(notebook_name, base_path='notebooks/'):
+        notebook_path = os.path.join(os.getcwd(), base_path, notebook_name + '.ipynb')
+        output_base_directory = os.path.join(os.getcwd(), base_path, 'output')
+        output_path = os.path.join(output_base_directory, notebook_name + '.output.ipynb')
+
+        start_time = time.time()
```

Comment:
I believe `time.perf_counter()` is preferred here. From https://www.webucator.com/blog/2015/08/python-clocks-explained/

> time.time() should NOT be used for comparing relative times. It’s not reliable because it’s adjustable.